contracts/ethereal-delta/src/lib.rs

```diff
@@ -83,12 +93,13 @@ mod delta {
       self.power_delta.take_all()
     }
 
-    pub fn deposit(&mut self, input: Bucket) {
+    pub fn deposit(&mut self, input: Bucket) -> Option<Bucket> {
```

Comment:
changing this would break Alpha's assumptions lol 
in general try not to change the behavior/signatures without verifying how the entire system depends on them

i.e. this function is meant to never fail and make a new resource vault if it doesn't exist